drivers/st7735/Kconfig

```diff
@@ -22,19 +22,19 @@ config HAVE_ST7735
     help
       Indicates that an ST7735 display is present.
 
+config HAVE_ST7789
```

Comment:
This seems unrelated and actually not needed. The pattern here was to group driver module variants  module with their `HAS_VARIANT` just after. With that change it looks messed up.

Maybe a better approach is to use:

```kconfig
config MODULE_VARIANT1
    ...

config MODULE_VARIANT2
    ...

config HAS_VARIANT1
    ...

config HAS_VARIANT2
    ...
```

Reply:
It is from the history. When @maribu tried to fix master compilation in CI, he added
```diff
  config MODULE_ST7789
-     bool "ST7789 display driver"
-     select MODULE_ST7735    bool
+     depends on HAVE_ST7789
+     default y if MODULE_ST7735
+     help
        ST7789 display driver
```
according to my suggestion. This was the commit cb0aba6669183fc403910d3dc578a9bb01ca746f in this PR. When I rebased, the order has been changed.

Regarding the order, my feeling was that we define a `HAVE_*` symbol before it is used in `MODULE_*`. I can change it if you want.

Reply:
I have changed the order.

Why is this fix not correct? `MODULE_ST7789` is only available if the board enables `HAVE_ST7789`. The `MODULE_ST7789` has to be enabled by default if `MODULE_ST7735` is enabled. This corresponds to makefile dependencies.

Reply:
"HAVE_ST7759" indeed tells the board has it so it can be pulled in automatically. But I manually plug the device to a dev kit and want to pull in myself, I'll also have to pullin the HAVE_ symbol which is not needed for other drivers AFAICT.

Reply:
> But I manually plug the device to a dev kit and want to pull in myself, I'll also have to pullin the HAVE_ symbol

Yes, indeed. Unfortunatly, there was no other a approach to enable the `st7789` variant if the board pulls in the module in makefile dependencies. This should become obsolte once the driver varaiants are splitted.